DESIGN.md

```diff
@@ -0,0 +1,169 @@
+  * status.py - make it easy for the user to see the status of the simulated bitcoin network
+5. Resource configuration pipeline
+  * admin.py - copy configurations for *resources* such as namespaces and put them in the user's directory
+  * deploy.py - take configurations for *resources* and put them into the Kubernetes cluster
+  * network.py - copy *resources* to the users Warnet directory
+  * namespaces.py - copy *resources* to the users Warnet directory; interact with namespaces in the cluster
+6. User interaction
+  * main.py - provide the interface for the `warnet` command line program
+7. Host computer
+  * process.py - provides a way to run commands on the user's host computer
+8. Externalized configuration
+  * constants.py - this holds values which occur repeatedly in the code base
+
+### Overview of test
+The test_base.py file forms the basis of the *test* section. Each test uses *TestBase* which controls the test running framework.
+
+### Operating in the network with "scenarios"/Commanders
+Warnet includes the capability to run "scenarios" on the network. These are python files which can be found in *resources/scenarios*, or copied by default into a new project directory.
+
+These scenarios use a base class called `Commander` which ultimately inherits from the Bitcoin Test Framework. This means that scenarios can be written and controlled using the familiar Bitcoin Core functional test language. The `self.nodes[index]` property has been patched to automatically direct commands to a running bitcoin node of the same index in the network.
+
+Once a scenario has been written, it can be loaded into the cluster and run using `warnet run <file>`.
+
+### The resources configuration pipeline - an example
+It is important to focus on the pipeline that takes *resources*, copies them into user directories, and translates them into Kubernetes objects. To make this possible and to achieve a more stateless configuration, Warnet uses Helm which provides templating for Kubernetes configuration files.
+
+Looking more closely at the *resources* section, for example, we can focus in on the *namespaces* directory. Inside, there is an example *namespaces.yaml* and *namespace-defaults.yaml* file. These configuration files are provided to the user when the `warnet admin init` command is invoked. This provides the user the opportunity to change those configuration files by specifying a set of participants who will have access to the simulated Bitcoin network. When `warnet deploy [namespaces_folder]` command is run by the user, it will apply the configuration data to the Helm chart found in the *charts* directory of the *resources* section. The Helm chart acts as a template through which the user's configuration data is applied. In this way, there is a pipeline which starts with the user's Stateful Data which is then piped through the Helm templating system, and then is applied to the Kubernetes cluster.
```

Comment:
Worth mentioning that these are under the charts directory? Would bitcoincore be the better example here? It's more core and will always be used whereas namespaces might not?